lib/fluent/plugin/in_tail.rb

```diff
@@ -499,23 +499,26 @@ def update_watcher(target_info, pe)
         end
       end
 
-      rotated_tw = @tails[path]
```

Comment:
NOTE: `@tails[path]` should point to the caller of this function ( = self-watcher).
So I pass self-watcher directly to this function.

This way, we can avoid the impact of `refresh_watcher` being executed first and `@tails` being overwritten.

Reply:
> NOTE: @tails[path] should point to the caller of this function ( = self-watcher).
> So I pass self-watcher directly to this function.

Maybe this is a good idea to prevent closing a new watcher wrongly.
Currently I'm checking this patch as code level and @kattz-kawa is evaluating this patch on our env.

Reply:
I agree with your idea too.